### ISSUE TITLE
Genre List Detail Page BSON Type Error 

### ISSUE THREAD

**Issue by jelugbadebo** [mdn/express-locallibrary-tutorial#135]
Following the MDN Web Docs server side Local Library Tutorial, for the Genre Detail Page, ([Genre Detail Page](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/Displaying_data/Genre_detail_page), the tutorial mentions a mongoose error coming from the req.params.id.

The tutorial recommends that we use mongoose.Types.ObjectId() to convert the id to a type that can be used. I implemented the mongoose.Types.ObjectId() code, but am now getting a different error: “BSONTypeError: Argument passed in must be a string of 12 bytes or a string of 24 hex characters or an integer”.

Does anyone know the fix for the BSON Type Error?

**Comment by hamishwillee** [comment]
Never seen it before.  Can you reproduce the problem with THIS project by following the instructions here: https://github.com/mdn/express-locallibrary-tutorial#quick-start

If so, in your version, do you have the same dependencies as here: This project https://github.com/mdn/express-locallibrary-tutorial/blob/master/package.json

**Comment by jelugbadebo** [comment]
Hi Hamish,

So to debug, I went through dependency-by-dependency and found the culprit.  

I had a “core-js” dependency installed, which was installed by default when I set up the express app framework at the very beginning of the Express/Node sever side tutorial.  This package is not used in the finished example MDN has on Github.  I removed the “core-js” package and voila, problem solved.  

I think I wasted a good 2-3 days trying to figure this out before I emailed you.

Lesson learned.

Thanks!
Johnson

**Comment by jelugbadebo** [comment]
So to debug, I went through dependency-by-dependency and found the culprit.

I had a “core-js” dependency installed, which was installed by default when I set up the express app framework at the very beginning of the Express/Node sever side tutorial. This package is not used in the finished example MDN has on Github. I removed the “core-js” package and voila, problem solved.

I think I wasted a good 2-3 days trying to figure this out before I emailed you.

Lesson learned.

Thanks!
Johnson

**Comment by jelugbadebo** [comment]
So to debug, I went through dependency-by-dependency and found the culprit.

I had a “core-js” dependency installed, which was installed by default when I set up the express app framework at the very beginning of the Express/Node sever side tutorial. This package is not used in the finished example MDN has on Github. I removed the “core-js” package and voila, problem solved.

I think I wasted a good 2-3 days trying to figure this out before I emailed you.

Lesson learned.

Thanks!
Johnson
